sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSourceLog.scala

```diff
@@ -122,8 +123,35 @@ class FileStreamSourceLog(
     }
     batches
   }
+
+  def restore(): Array[FileEntry] = {
```

Comment:
To not touch existing semantic of `allFiles()`, I simply add a new method to cover the new semantic. I'll just override `allFiles()` if it's preferred.